sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -445,6 +459,7 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 		links[id].name = "dmic01";
 		links[id].cpus = &cpus[id];
 		links[id].cpus->dai_name = "DMIC01 Pin";
+		links[id].init = dmic_init;
```

Comment:
so what if we use the dmic16k first, the init would not be run? Seems like this is a card init rather than a link init anyways?

Reply:
@plbossart The init will still run even we use dmic16k first, and it is still valid. Both "dmic01" and "dmic16k" use the same codec component. Use link init is also consistent with other machine drivers. And we only set it in one dmic dailink which will guarantee the init will be invoked only once.

Reply:
Sorry @bardliao I don't see how a dai_link init for one link will be called when *another* dailink is initialized. There's something fishy here.
Or are you saying that it doesn't matter, all links are initialized first before one of them is used?

Reply:
@plbossart Sorry I thought your "use" means "write". So actually I was trying to explain that it will work even if we set .init to "dmic16k" dai link.
Back to your question. The init will be invoked when a dai link is registered. So yes, it will be initialized before any of them is used.

Reply:
ok, got it.